guacamole-ext/src/main/java/org/apache/guacamole/net/auth/ActivityRecordSet.java

```diff
@@ -56,6 +56,24 @@
      *      If an error occurs while retrieving the records within this set.
      */
     Collection<RecordType> asCollection() throws GuacamoleException;
+    
+    /**
+     * Returns records within this set for the item having the specified
+     * identifier as a standard Collection.
+     *
+     * @param identifier
+     *     The identifier of the underlying item that the collection
+     *     should be limited to.
+     * 
+     * @return
+     *     A collection containing records within this set for the specified
+     *     identifier.
+     *
+     * @throws GuacamoleException
+     *     If an error occurs while retrieving the records within this set.
+     */
+    Collection<RecordType> asCollection(String identifier)
+            throws GuacamoleException;
```

Comment:
A couple things:

* To continue maintaining backward compatibility, default implementations need to be provided for new functions added to existing interfaces within guacamole-ext.
* As defined, this will be ambiguous as to which type of object the identifier refers to. The records within the collection may point to multiple types of objects. For example, a connection record points to both a user and a connection, and may also point to a sharing profile.
* The general pattern for `ActivityRecordSet` is a set of functions which can be layered on top of each other by virtue of those functions each returning `ActivityRecordSet`, similar to the various "builder" objects we see in JAX-RS. Keeping with that pattern, I think this would be best implemented not as new version of `asCollection()`, but as some other function that allows the set to be narrowed to records containing a particular object having a particular identifier.

   I could imagine a new version of `contains()` which accepts an `Identifiable` rather than a `String`, or perhaps multiple `contains()` for `Connection`, `User`, `SharingProfile`, etc.

Reply:
Another possibility could be to add something like `getConnectionHistory()` and `getUserHistory()` (returning `ActivityRecordSet` like their `UserContext` versions) to `Connection` and `User` respectively, deprecating the old `getHistory()`.

That seems clean but would also lend itself well to default implementations (could just wrap `getHistory()` in some sort of basic `ActivityRecordSet` implementation).

Reply:
ok

Reply:
Okay, I have taken a wild stab at this - let me know how it looks, now.